### PR TITLE
feat: add filters to the query preview request

### PR DESCRIPTION
**Pull request by lauramargar**
# Pull request template
 
Currently, the query preview component accept 2 params, title&query, so it creates it retrieves the products and show them in a viz, an sliding panel as default.

It would be great if we could add filters & extra params to the search request, now extra params are developed, so we also have to add filters. These params could help to narrow the results, that could be very interesting for some customers.

## Motivation and context

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link: [EMP-2141](https://searchbroker.atlassian.net/browse/EMP-2141)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

## How has this been tested?

At `Home.vue` file add or remove filters in `queriesPreviewInfo` var. The carrousels of query preview should be filtered.

[EMP-2141]: https://searchbroker.atlassian.net/browse/EMP-2141?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/x-components/src/x-modules/queries-preview/components/query-preview.vue [hunk 1]
```diff
@@ -161,11 +162,19 @@
...
         location: this.location
       });
-
+      const filtersApply = this.queryPreviewInfo.filters?.reduce((filtersList, filterId) => {
+        const facetId = filterId.split(':')[0];
+        const rawFilter = createRawFilter(filterId);
+        filtersList[facetId] = filtersList[facetId]
+          ? filtersList[facetId].concat(rawFilter)
+          : [rawFilter];
+        return filtersList;
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/queries-preview/components/query-preview.vue:171]: Add space here for clarity.

```suggestion

        return filtersList;
```

File: packages/x-components/src/x-modules/queries-preview/components/query-preview.vue [hunk 2]
```diff
@@ -161,11 +162,19 @@
         feature: this.queryFeature,
         location: this.location
       });
-
+      const filtersApply = this.queryPreviewInfo.filters?.reduce((filtersList, filterId) => {
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/queries-preview/components/query-preview.vue:165]:
```suggestion
      const filters = this.queryPreviewInfo.filters?.reduce((filtersList, filterId) => {
```

File: packages/x-components/src/x-modules/queries-preview/components/query-preview.vue [hunk 3]
```diff
@@ -161,11 +162,19 @@
...
+        filtersList[facetId] = filtersList[facetId]
+          ? filtersList[facetId].concat(rawFilter)
+          : [rawFilter];
+        return filtersList;
+      }, {} as Record<string, Filter[]>);
       return {
         query: this.queryPreviewInfo.query,
         rows: this.config.maxItemsToRequest,
         extraParams: { ...this.params, ...this.queryPreviewInfo.extraParams },
+        filters: filtersApply,
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/queries-preview/components/query-preview.vue:177]:
```suggestion
        filters,
```

File: packages/x-components/src/views/home/Home.vue [hunk 4]
```diff
@@ -626,10 +626,12 @@
...
       {
         query: 'cortina',
-        extraParams: { store: 'Gijón' }
+        extraParams: { store: 'Gijón' },
+        filters: ['categoryIds:66dd06d9f']
       },
       {
-        query: 'marni summer dress'
+        query: 'marni summer dress',
+        filters: ['categoryIds:5b612edb5', 'fit:regular']
```
Comment (herrardo) [inline packages/x-components/src/views/home/Home.vue:634]: I would change the example to see it work as a real example:

```suggestion
        query: 'summer dress',
        filters: ['categoryIds:5b612edb5', 'brand:marni']
```

File: packages/x-components/src/x-modules/queries-preview/components/query-preview.vue [hunk 5]
```diff
@@ -161,11 +162,19 @@
...
-
+      const filtersApply = this.queryPreviewInfo.filters?.reduce((filtersList, filterId) => {
+        const facetId = filterId.split(':')[0];
+        const rawFilter = createRawFilter(filterId);
+        filtersList[facetId] = filtersList[facetId]
+          ? filtersList[facetId].concat(rawFilter)
+          : [rawFilter];
+        return filtersList;
+      }, {} as Record<string, Filter[]>);
       return {
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/queries-preview/components/query-preview.vue:173]: Add space here for clarity.

```suggestion

      return {
```

File: packages/x-components/src/x-modules/queries-preview/components/query-preview.vue
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/queries-preview/components/query-preview.vue:1]: In this component's test 'sends the `QueryPreviewRequestUpdated` event' add filters to the `queryPreviewInfo` and check that the `queryPreviewRequest` has them.

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Pull request merged**